### ISSUE TITLE
framework: esp-idf throws Syslog.h not found

### ISSUE THREAD

**Issue by seganku** [TheStaticTurtle/esphome_syslog#12]
I am building ESPHome Bluetooth Proxies for [Sonoff Mini R4](https://digiblur.com/wiki/devices/relays/sonoff-mini-r4/) (ESP32) switches.  The [ESPHome link](https://esphome.io/components/bluetooth_proxy.html) seems quite adamant that we use `framework: {type: esp-idf}`.  When I switch from `{type: arduino}` to `{type: esp-idf}`, I start getting this error:
```
In file included from src/esphome/components/syslog/syslog_component.cpp:1:
src/esphome/components/syslog/syslog_component.h:9:10: fatal error: Syslog.h: No such file or directory
```

Example config:
```yaml
esphome:
  name: syslog-esp-idf
esp32:
  board: esp32dev
  framework:
    type: esp-idf
logger:
syslog:
  ip_address: !secret syslog_ip_address
api:
ota:
wifi:
  ssid: !secret wifi_ssid
  password: !secret wifi_password
```

**Comment by TheStaticTurtle** [comment]
Hi, please excuse the delayed response.

My component is merely a wrapper that connects https://github.com/arcao/Syslog and esphome.
I'm not certain how esphome handles dependencies and how/if it would include a library that was designed for the Arduino framework.

However, the library is simple, and you might be able to port it to esp-idf compatible code. The lib is well-designed and most of the code can stay the same, you would only need to look at these functions to use the proper methods to send UDP messages:
 - `inline bool Syslog::_sendLog(uint16_t pri, const char *message)`
 - `inline bool Syslog::_sendLog(uint16_t pri, const __FlashStringHelper *message) `


Samuel.

**Comment by AJolly** [comment]
Ah, thats too bad :/

**Comment by dwmw2** [comment]
Should be fixed by #17
